rasa/core/training/training.py

```diff
@@ -65,7 +65,7 @@ def create_action_fingerprints(
 
     # take into account only featurized slots
     featurized_slots = {slot.name for slot in domain.slots if slot.has_features()}
-    action_fingerprints = defaultdict(dict)
+    action_fingerprints: Dict[Text, Dict[Text, List[Text]]] = defaultdict(dict)
```

Comment:
Can we use `typing.DefaultDict` with the python versions that we support?

Reply:
[it's actually deprecated](https://docs.python.org/3/library/typing.html#typing.DefaultDict), using `defaultdict` directly seems to work locally, I pushed 702d3da1108 to see if it works well in the CI

Reply:
Oh yes, those additions that get rid of "typing." are nice :) But did we merge 3.9 already? Do we run those checks in other python versions also? 🤔

Reply:
yes we do, but we have a couple of backwards compatibility modules that get installed, to make sure we can use the latest typing good stuff